Add deploy task for supported platforms

Luca-Lauro opened the pull request:
- update packagers:
  - add deploy functions
  - replace js2xmlparser with xml2js for bi-directional conversion in Orsay packager
  - refactor getManualOrsayConfData to use xml2js
- deploy task for sectv-orsay tested with Orsay (Legacy) Smart TV

Luca-Lauro committed:
Add deploy task for supported platforms

- update packagers:
  - add deploy functions
  - replace js2xmlparser with xml2js for bi-directional conversion
  - refactor getManualOrsayConfData to use xml2js
- deploy task for sectv-orsay tested with Orsay (Legacy) Smart TV

Luca-Lauro committed:
Improved widgetlist management for Orsay platform

Now if you re-deploy an existing widget without deleting it from widgetlist.xml, it will be overwritten instead of being added another time.

prozanne commented on tasks/packager/sectv-tizen.js:
```diff
@@ -368,5 +368,40 @@ module.exports = {
+        if(fs.existsSync(userConfPath)){
+            var userData = JSON.parse(fs.readFileSync(userConfPath));
+
+            if(userData.hasOwnProperty('orsay')){
```
I think there is a typo, it should be 'tizen' I think.

Luca-Lauro replied:
yeah, you're right.
Thanks for pointing that out.

prozanne commented on tasks/packager/tv-webos.js:
```diff
@@ -421,5 +421,41 @@ module.exports = {
+        if(fs.existsSync(userConfPath)){
+            var userData = JSON.parse(fs.readFileSync(userConfPath));
+
+            if(userData.hasOwnProperty('orsay')){
```
Here as well.

Luca-Lauro replied:
Same here.

Luca-Lauro committed:
Typo correction

Correction of typo oversights in:
- tasks/packager/sectv-tizen.js
- tasks/packager/tv-webos.js

prozanne commented on tasks/packager/sectv-tizen.js:
```diff
@@ -368,5 +368,40 @@ module.exports = {
+        console.log('\nStart deploying Tizen Samsung Smart TV Platform apps......');
+
+        var { spawn } = require('child_process');
+        var wgtPath = data.wgtPath || './platform/tizen/build/app.wgt';
```
the default dest path should be ./platform/sectv-tizen/build/app.wgt

It's sorry for saying that,
There are some parts of this PR that are questionable as to whether it works properly, so It's hard to be accepted for now.
Please test all platforms and make sure that they all work well.
Please consider to attach the working well screenshots for each platforms,
Then I will accept this PR.

Luca-Lauro replied:
You're right.
As reported, the only tested platform that works so far is Orsay.
I'd say to wait for someone to test the other platforms first.
Thanks for your attention.
